torchvision/datasets/lsun.py

```diff
@@ -96,13 +101,12 @@ def __init__(self, db_path, classes='train',
                                      'Options are: ' + str(dset_opts)))
         else:
             raise(ValueError('Unknown option for classes'))
-        self.classes = classes
```

Comment:
This doesn't look right: there is a few operations that are performed just before that modify `classes`, and you don't take that into account when moving `self.classes` up.